File: qubit.cpp
#include "qubit.h"

Qubit::Qubit(const Matrix& matrix, int numQubits, const std::string& name) {
    this->data = matrix;
    this->numQubits = numQubits;
    this->name = name;
}

Qubit::Qubit() {
    this->data = Matrix();
    this->numQubits = 0;
    this->name = "";
}

Qubit::Qubit(const Matrix& matrix) {
    this->data = matrix;
    this->numQubits = matrix.rows;
    this->name = "";
}


Qubit Qubit::applyGate(Gate& gate){
    return *(new Qubit(this->data * gate.gateMatrix));
}

File: src/javaProject/Qubit.java
public class Qubit extends Matrix{

    public Qubit(){
        this.cols = 0;
        this.rows = 0;
        this.data = new ComplexNumber[0][0];
    }
    
    public Qubit(int rows){
        this.rows = rows;
        this.cols = 1;
        this.data = new ComplexNumber[rows][1];
    }

    public Qubit(int rows, ComplexNumber[][] data){
        this.rows = rows;
        this.cols = 1;
        this.data = data;
    }

}